### DISCUSSION TITLE
Split dependency updates by file

### DISCUSSION THREAD

**Discussion by cdenneen** [renovatebot/renovate#28505] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Currently I've been using renovate with much success however I have a requirement I need to fulfill.
For example if I have a HelmRelease (flux resource) of say external-dns. I need to have the chart version in the baseline HelmRelease but I also want to have the version in an environment specific Kustomization patch or even a cluster specific Kustomization patch.

Currently when Renovate finds these it will provide 1 PR bumping the chart version for all of them. What I'd like to have is these broken into separate commits/PRs due to the fact that I'd like to be able to test these changes on a test cluster first, then rollout to dev, qa, production (at which time we can merge the change into base).

How can this be achieved?

**Comment by rarkins** [comment, marked as answer]
This can be achieved by configuring `additionalBranchPrefix` for those packages to be something like `{{parentDir}}-`. There are some past questions/examples if you search for those terms.

**Comment by cdenneen** [comment]
So this helps by creating separate branches which is great however the title of the MR remains the same `chore(deps): update terraform helm to v2.13.1` so it won't open a MR for each one and even after one is merged the old one is merge/closed so it won't open another ... so the "prefix" needs to be added to the MR name as well I fear.

**Reply by rarkins** [reply]
Are you sure about that or assuming?

**Reply by cdenneen** [reply]
I was able to get this to work by setting `commitMessageAction: "update {{parentDir}}`

**Reply by cdenneen** [reply]
I am sure because watching renovate logs I could see it opened the branch but wouldn't open the PR due to existing one.

**Reply by rarkins** [reply]
Next time create a reproduction if you want us to look into it. Renovate bases its "should it open or not" based on the combination of branch name + PR title, so an existing PR with same title and different branch name shouldn't cause any blocking. You may have misunderstood the logs. Please copy them here if you have them.
